astarte_device.c

```diff
@@ -927,6 +929,10 @@ static void on_incoming(
         return;
     }
 
+    if (!device->unset_data_event_callback) {
+        ESP_LOGW(TAG, "unset_data_event_callback not set");
```

Comment:
I think here maybe is too much logging a warning, since there is still a lot of old devices that are not using it (and they likely not need it). The other LOGE we have is enough.

Reply:
++